oauthenticator/oauth2.py

```diff
@@ -261,6 +267,16 @@ def _userdata_url_default(self):
         """,
     )
 
+    @default("extra_authorize_params")
+    def _extra_authorize_params(self):
+        return os.environ.get("OAUTH2_EXTRA_AUTHORIZE_PARAMS", {})
```

Comment:
The environment variable will be a string, so you'll somehow need to convert it to a `dict`. Might be easier to remove it?

Reply:
That's a good point, I'll take this out.

Reply:
@NickolausDS instead of taking it out, why not do something like this?

https://github.com/jupyterhub/oauthenticator/blob/master/oauthenticator/google.py#L87-L95

Reply:
That's a good idea, but it will be a bit more difficult to parse a dict rather than a list. For a dict type, encoding as a JSON string would probably be easiest. However, JSON is syntax heavy and uncomfortable to cram into an environment string. 

Unless you think there's a need for it, I think the most common use-case would be providing these extra params as part of the jupyterhub.py config. If we did want to add an additional config option per-provider, such as `REFRESH_TOKENS=True` for `google.py`, that could be added as a separate property that updates `extra_authorize_params`. Each provider seems to be pretty unique for which extra parameters it supports (refresh tokens seem to be the most common, but not all providers support them), so this might be the best middle ground.

Reply:
I was thinking something like `:` separates key from value, and `;` separates each entry. But you are right this is not necessary at all.

Reply:
👍 It should be easy enough to add this in future if there's demand, it's much hard to remove/change maintain something once it's added.